creating question model, binding it to the multiple choice page, crea…

skogmanpe40 opened the pull request:
started the process of being able to populate the questions with data from the database

skogmanpe40 committed:
creating question model, binding it to the multiple choice page, create business logic layer

benderje26 commented on Models/Question.cs:
My personal opinion on this would be to instead of fixing the number of possible questions, make the questions either an array or a list, and have the correct answer be the index instead of a string. Consider that, and maybe we should consider instead of having a boolean asking if it is multiple choice, maybe we should have two classes - MultipleChoiceQuestion and InputQuestion or something along those lines which both extend Question. Let me know what you think

benderje26 reviewed (changes requested):
Look at what I suggested and let me know what you think

skogmanpe40 committed:
make fixes requested by Jeremiah and some nav

Made the changes to question model given by Jeremiah since I
agree that those changes will be benefical for run time.
Basic naviagtion between questions and questions stats added.

skogmanpe40 committed:
Merge branch 'master' into peter

benderje26 committed:
Update models for questions and databases

- Update MultipleChoiceQuestion and FillBlankQuestion
- Update quiz model to match new design
- Update business logic to get questions (will get changed again in future sprint)
- Add ability to get questions from IDatabase (will also change in future sprint)

benderje26 reviewed (approved):
This is not stable right now but we need this in main for other people to work on stuff

pachia-lee commented on Models/FillBlankQuestion.cs:
```diff
@@ -0,0 +1,23 @@
+    public override bool HasCorrectAnswer() => correntAnswers != null;
+
+    public override bool IsCorrect() {
+        string[] possibilities = correntAnswers ?? [];
```
did you mean correctAnswers?

pachia-lee commented on Models/Question.cs:
```diff
@@ -0,0 +1,131 @@
+public class Question : INotifyPropertyChanged
+{
+    String problem;
+    String correntAnswer;
```
Not sure if my other comment disappeared or whatever, but did you mean correctAnswer here..

skogmanpe40 replied:
Good catch, I went ahead and fixed the spelling mistake

pachia-lee reviewed (commented):
Just had a comment about correctAnswer, might just be a spelling mistake

skogmanpe40 committed:
Fix the small spelling mistake

Fix the mistake of correntAnswer to correctAnswer

pachia-lee commented on Models/Quiz.cs:
```diff
@@ -1,14 +1,32 @@
-    public Quiz(string title, DateTime lastActivated) {
-        this.Title = title;
-        this.LastActivated = lastActivated;
+public class Quiz(string title, DateTime? lastActivated, DateTime? dateCreated, long? time, List<Question> questions) {
```
Looks like this constructor breaks other parts of the app like in the QuizHistory.xaml.cs file.

skogmanpe40 committed:
Add previous and next button

Still need to figure out how to move to the next or previous question in the
question array when the next or previous button is pressed.

skogmanpe40 committed:
Finish navigation between quiz questions

Can now naviagte between quiz questions. If the next questions is
multiple choice it will pull up the multiple choice page and if it is
fill blank the fill blank page will pull up.

pachia-lee reviewed (commented):
During a live quiz, do you think people would have the need to view the previous question? Also the buttons are quite close together..

skogmanpe40 committed:
Make the previous and next buttons farther apart

Made the buttons farther apart as requested by Pachia, made the page
look a lot better.

skogmanpe40 committed:
Add comments to methods where needed
